strapi/strapi
# feat(content-releases): update cron scheduling when release scheduledAt is changed

### What does it do?

* When editing a release, you can edit the scheduledAt date or remove it. Then, we update the scheduledJobs accordingly.
* Additionally, I took this opportunity to fix a bug: You can't have two pending releases with the same name. This validation is enforced when creating, but not when updating. I have added this validation for updating in this PR as well.

## Testing

### Enabling the future flags

The scheduling functionality is implemented through future flags. To begin, ensure the future flag is set to true. Navigate to `config/features.js` and include `contentReleasesScheduling` as an option within your future object. Create the future object if it does not exist.

Example:
```js
// config/features.js

module.exports = ({ env }) => ({
  future: {
    contentReleasesScheduling: true,
  },
});

```

### How to test it

Note: You need to get all the scheduledJobs from scheduling service (There is no a way in UI to see this, so you would need to go to the console)

1. Create a new release with a scheduled date.
2. Try to edit the release, changing the scheduled date and it should change.
3. Try to remove the scheduled date by passing `scheduledAt` as null
